libbpf-rs/src/map.rs

```diff
@@ -476,6 +476,22 @@ impl Map {
             ptr: std::ptr::null_mut(),
         })
     }
+
+    /// Attach a struct ops map
+    pub fn attach_struct_ops(&mut self) -> Result<Link> {
+        if self.map_type() != MapType::StructOps {
+            return Err(Error::InvalidInput(format!(
+                "Invalid map type ({}) for attach_struct_ops()",
+                self.map_type(),
+            )));
+        }
+
+        let ret = unsafe { libbpf_sys::bpf_map__attach_struct_ops(self.ptr) };
+        if ret.is_null() {
+            return Err(Error::System(nix::errno::errno()));
+        }
+        Ok(Link::new(ret))
```

Comment:
I think you should use `libbpf_get_error()` instead.

Reply:
should libbpf-rs use libbpf 1.0 strict mode always?

Reply:
Not sure where and how to place `libbpf_set_strict_mode()` in the codebase in a Rust way.

Reply:
We can probably add it before bpf_object__open() call. But that's separate from this change.